app/src/main/java/uk/ac/bath/petmatch/PetEditActivity.java

```diff
@@ -32,17 +44,75 @@ protected void onCreate(Bundle savedInstanceState) {
         final EditText editDesc = (EditText)findViewById(R.id.petDesc);
         editDesc.setText(petClicked.getDescription(), TextView.BufferType.EDITABLE);
 
+        ImageView petImage = (ImageView)findViewById(R.id.petImage);
+        String id = petClicked.getImage();
+        int resID;
+        try {
+            resID = (int) R.mipmap.class.getField(id).get(null);
+        } catch (Exception e) {
+            Log.e("PetGridAdapter", e.getMessage());
+            resID = R.mipmap.dumbledore;
+        }
+        petImage.setImageResource(resID);
+        //petImageView.setImageResource(R.mipmap.dumbledore);
+
+        Spinner spinner = (Spinner) findViewById(R.id.breeds_spinner);
+        // Create an ArrayAdapter using the string array and a default spinner layout
+
+        //Get list of breeds in a given breed type
+        String[] breeds = getHelper().petBreeds.getArrayForType();
+        final String[] spinnerBreeds = new String[breeds.length +1];
+        for (int i = 0; i < breeds.length; i++) {
+            spinnerBreeds[i] = breeds[i];
+        }
+        spinnerBreeds[breeds.length] = petClicked.getBreed().getTitle(); //add an "empty" choice option
```

Comment:
This is not adding an empty choice, as per your comment, it is rewriting one of the options with the current breed, thus making the one breed appear twice. This is NOT how you set a spinner default value.